src/routes/firmware.ts

```diff
@@ -0,0 +1,17 @@
+import { Router } from 'express';
+import FirmwareManager from 'util/firmware';
+const router = Router();
+
+const firmwareManager = new FirmwareManager();
+
+router.get('/install', async (req, res) => {
+  const firmwareFile = req?.body?.fileName || '';
+  const resp = firmwareManager.installFirmware(firmwareFile);
+  res.json(resp);
+});
+
+router.get('/progress', async (req, res) => {
+  res.json({ isRunning: firmwareManager.getMessage(), errorSeen: firmwareManager.getErrorSeen() });
```

Comment:
sorry but I think it's easier if it return the json. Otherwise if I use somewhere else I would need to duplicate the same code
`getProgressMessage() --> return { isRunning, error }`

lets keep error (not error seen to keep api consistency

Reply:
so the router response is
`res.json(getInstallationProgress());`

Reply:
and wherever else I use I can just call `getInstallationProgress` and we have the same api response type (on the socket for example)